### PR TITLE
vacuum metrics and force sync dst files

### PR DESCRIPTION
**Pull request by kmlebedev**
# What problem are we solving?

I would like to see the regression of the vacuum if it is when updating seaweedfs
https://github.com/seaweedfs/seaweedfs/issues/3837

# How are we solving the problem?

add vacuum error count and  seconds metrics
Screenshot 2022-10-12 at 17 33 25

# How is the PR tested?

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/storage/volume_vacuum.go [hunk 1]
```diff
@@ -265,13 +268,20 @@ func (v *Volume) makeupDiff(newDatFileName, newIdxFileName, oldDatFileName, oldI
 		return fmt.Errorf("open dat file %s failed: %v", newDatFileName, err)
 	}
 	dstDatBackend := backend.NewDiskFile(dst)
-	defer dstDatBackend.Close()
+	defer func() {
+		dstDatBackend.Sync()
```
Comment (chrislusf) [inline weed/storage/volume_vacuum.go]: consider moving the sync() into close()

**Comment by guol-fnst** [comment]
I also found that current vacuum speed is much slower than before when I running test.

**Comment by chrislusf** [comment]
> I also found that current vacuum speed is much slower than before when I running test.

@guol-fnst You are comparing which two versions?

**Comment by kmlebedev** [comment]
> I also found that current vacuum speed is much slower than before when I running test.

This is to be expected due to direct-to-disk writing being enabled.
If speed is more important to you than data integrity, then you can move this behind the vacuum_fsync option

**Comment by guol-fnst** [comment]
> > I also found that current vacuum speed is much slower than before when I running test.
> 
> @guol-fnst You are comparing which two versions? What is your OS?

Actually I found this problem when I tested my laster PR (https://github.com/seaweedfs/seaweedfs/pull/3830)
Besides I found the same problem when I tested my another PR(https://github.com/seaweedfs/seaweedfs/issues/3443)
And a related PR was reverted.

**Comment by guol-fnst** [comment]
> 

Yes, I totally undanstand your concerns and know why it gets slower.
I just share this information here.
And thanks for your explanation.

**Comment by guol-fnst** [comment]
> > I also found that current vacuum speed is much slower than before when I running test.
> 
> @guol-fnst You are comparing which two versions? What is your OS?

Almost forgot. Centos-7.2

And I think it's necessary to add a CI/CD for performance  test for each PR expecially when you are sensitive to performance

**Pull request merged**